src/Microsoft.ML.FastTree/RegressionTree.cs

```diff
@@ -0,0 +1,229 @@
+        /// </summary>
+        public IReadOnlyList<int> GetCategoricalSplitFeaturesAt(int nodeIndex)
+        {
+            if (nodeIndex < 0 || nodeIndex >= NumNodes)
+                throw Contracts.Except($"The input index, {nodeIndex}, is invalid. Its valid range is from 0 (inclusive) to {NumNodes} (exclusive).");
+
+            if (_tree.CategoricalSplitFeatures == null || _tree.CategoricalSplitFeatures[nodeIndex] == null)
+                return new List<int>(); // Zero-length vector.
+            else
+                return _tree.CategoricalSplitFeatures[nodeIndex];
+        }
+        /// <summary>
+        /// Return categorical thresholds' range used at node indexed by nodeIndex. A categorical split at node indexed
+        /// by nodeIndex can consider multiple consecutive input features at one time; their range is specified by
+        /// <see cref="GetCategoricalCategoricalSplitFeatureRangeAt(int)"/>. The returned value is always a 2-element
+        /// array; its 1st element is the starting index and its 2nd element is the endining index of a feature segment.
+        /// The returned value is valid only if <see cref="CategoricalSplitFlags"/>[nodeIndex] is true.
+        /// </summary>
+        public IReadOnlyList<int> GetCategoricalCategoricalSplitFeatureRangeAt(int nodeIndex)
+        {
+            if (nodeIndex < 0 || nodeIndex >= NumNodes)
+                throw Contracts.Except($"The input node index, {nodeIndex}, is invalid. Its valid range is from 0 (inclusive) to {NumNodes} (exclusive).");
+
+            if (_tree.CategoricalSplitFeatureRanges == null || _tree.CategoricalSplitFeatureRanges[nodeIndex] == null)
+                return new List<int>(); // Zero-length vector.
+            else
+                return _tree.CategoricalSplitFeatureRanges[nodeIndex];
+        }
+
+        /// <summary>
+        /// Number of leaves in the tree. Note that <see cref="NumLeaves"/> does not take non-leaf nodes into account.
+        /// </summary>
+        public int NumLeaves => _tree.NumLeaves;
+
+        /// <summary>
+        /// Number of nodes in the tree. This doesn't include any leaves. For example, a tree with node0->node1,
+        /// node0->leaf3, node1->leaf1, node1->leaf2, <see cref="NumNodes"/> and <see cref="NumLeaves"/> should
+        /// be 2 and 3, respectively.
+        /// </summary>
+        // A visualization of the example mentioned in this doc string.
+        //         node0
+        //         /  \
+        //     node1 leaf3
+        //     /  \
+        // leaf1 leaf2
+        // The index of leaf starts with 1 because interally we use "-1" as the 1st leaf's index, "-2" for the 2nd leaf's index, and so on.
+        public int NumNodes => _tree.NumNodes;
+
+        internal RegressionTreeBase(InternalRegressionTree tree)
+        {
+            _tree = tree;
+
+            _lteChild = ImmutableArray.Create(_tree.LteChild, 0, _tree.NumNodes);
```

Comment:
>ImmutableArray [](start = 24, length = 14)

This can do for now, but FYI wrapping in an immutable array creates a *copy*. Not a deal breaker, just something you ought to keep in mind.